apps/system/js/app_install_manager.js

```diff
@@ -1,714 +1,718 @@
+      this.imeLayoutDialog = new ImeLayoutDialog();     
+
+      this.notifContainer =
+              document.getElementById('install-manager-notification-container');
+      this.appInfos = {};
+      this.setupQueue = [];
+      this.isSetupInProgress = false;
+      window.addEventListener('mozChromeEvent',
+        (function ai_handleChromeEvent(e) {
+        if (e.detail.type == 'webapps-ask-install') {
+          this.handleAppInstallPrompt(e.detail);
+        }
+        if (e.detail.type == 'webapps-ask-uninstall') {
+          this.handleAppUninstallPrompt(e.detail);
+        }
+      }).bind(this));
+
+      window.addEventListener('applicationinstall',
+        this.handleApplicationInstall.bind(this));
+
+      window.addEventListener('applicationuninstall',
+        this.handleApplicationUninstall.bind(this));
+
+      this.dialog.elements.installButton.onclick =
+        this.handleInstall.bind(this);
+      this.dialog.elements.cancelButton.onclick =
+        this.showInstallCancelDialog.bind(this);
+      this.installCancelDialog.elements.confirmCancelButton.onclick =
+        this.handleInstallCancel.bind(this);
+      this.installCancelDialog.elements.resumeButton.onclick =
+        this.hideInstallCancelDialog.bind(this);
+
+      this.notifContainer.onclick = this.showDownloadCancelDialog.bind(this);
+
+      this.downloadCancelDialog.elements.stopButton.onclick =
+        this.handleConfirmDownloadCancel.bind(this);
+      this.downloadCancelDialog.elements.continueButton.onclick =
+        this.handleCancelDownloadCancel.bind(this);
+
+      this.setupInstalledAppDialog.elements.cancelButton.onclick =
+        this.handleSetupCancelAction.bind(this);
+      this.setupInstalledAppDialog.elements.confirmButton.onclick =
+        this.handleSetupConfirmAction.bind(this);
+
+      this.imeLayoutDialog.elements.cancelButton.onclick =
+        this.hideIMEList.bind(this);
+      this.imeLayoutDialog.elements.confirmButton.onclick =
+        this.handleImeConfirmAction.bind(this);
+
+      // bind these handlers so that we can have only one instance and check
+      // them later on
+      ['handleDownloadSuccess',
+       'handleDownloadError',
+       'handleProgress',
+       'handleApplicationReady'
+      ].forEach(function(name) {
+        this[name] = this[name].bind(this);
+      }, this);
+
+      window.addEventListener('applicationready',
+          this.handleApplicationReady);
+
+      window.addEventListener('home', this.handleHomeButtonPressed.bind(this));
+    },
+
+    handleHomeButtonPressed: function ai_handleHomeButtonPressed(e) {
+      this.dialog.hide();
+      this.handleInstallCancel();
+
+      // hide IME setup dialog if presented
+      if (!this.setupInstalledAppDialog.element.hidden) {
+        this.handleSetupCancelAction();
+      }
 
-    var apps = e.detail.applications;
+      // hide IME layout list if presented
+      if (!this.imeLayoutDialog.element.hidden) {
```

Comment:
if (this.imeLayoutDialog && this.imeLayoutDialog.isActive()) {
  this.imeLayoutDialog.hide();
}